### PR TITLE
[BEAM-1188] Accept Multiple Verifiers in One IT

### PR DESCRIPTION
**Pull request by markflyhigh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

Each integration test can set multiple verifiers instead of one. This can help users to verify different aspects of pipeline result in one integration test when more verifiers will be provided in `apache_beam.tests.pipeline_verifiers.py`. User can also build own verifiers to meet their needs.

TODO:
More verifiers are on their way.

### CODE REVIEW

File: sdks/python/apache_beam/examples/wordcount_it_test.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ class WordCountIT(unittest.TestCase):
   @attr('IT')
   def test_wordcount_it(self):
     # Set extra options to the pipeline for test purpose
-    extra_opts = {'on_success_matcher': PipelineStateMatcher()}
```
Comment (lukecwik) [inline sdks/python/apache_beam/examples/wordcount_it_test.py]: Any reason not to use the all or any of pattern for matchers?

This allows the caller to choose whether all conditions need to be met, or some conditions or some combination of both.

Comment (markflyhigh) [inline sdks/python/apache_beam/examples/wordcount_it_test.py]: good to learn this pattern. 

However, I'm not sure how to use them for matchers. According to [doc](https://docs.python.org/2/library/functions.html#all), `any` and `all` check`True`/`False` elements, but for matcher(s) we need to call `assert_that` for verification.

**Comment by markflyhigh** [comment]
+R: @aaltay

File: sdks/python/apache_beam/examples/wordcount_it_test.py (hunk 1 above)
Comment (lukecwik) [inline sdks/python/apache_beam/examples/wordcount_it_test.py]: I was referring to [all_of and any_of](https://pypi.python.org/pypi/PyHamcrest) part of PyHamcrest

Comment (markflyhigh) [inline sdks/python/apache_beam/examples/wordcount_it_test.py]: right, Pyhamcrest `any_of` and `all_of` do exactly what I want in this PR, but in a cleaner way. Thanks for suggestion!

**Comment by markflyhigh** [comment]
Suggested by @lukecwik:

We can use `all_of` and `any_of` in Pyhamcrest to choose which conditions to be met, which is what I want to achieve in this PR. Also more logical matchers are provided in [Pyhamcrest](https://pypi.python.org/pypi/PyHamcrest). 

Will close this PR.

**Pull request closed without merging**